nautobot_secrets_providers/providers/keeper.py

```diff
@@ -0,0 +1,188 @@
+"""Secrets Provider for Keeper."""
+import os
+# from pathlib import Path
+# import base64
+# import json
+
+try:
+    from keeper_secrets_manager_core import SecretsManager
+    from keeper_secrets_manager_core.core import KSMCache
+    from keeper_secrets_manager_core.exceptions import KeeperError, KeeperAccessDenied
+    from keeper_secrets_manager_core.storage import FileKeyValueStorage #, InMemoryKeyValueStorage
+    # from keeper_secrets_manager_core.utils import get_totp_code
+except (ImportError, ModuleNotFoundError):
+    keeper = None
+
+from django import forms
+from django.conf import settings
+# from django.core.exceptions import ValidationError
+
+from nautobot.apps.secrets import exceptions, SecretsProvider
+from nautobot.utilities.forms import BootstrapMixin
+
+from .choices import KeeperTypeChoices
+
+
+__all__ = ("KeeperSecretsProvider")
+
+
+try:
+    plugins_config = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]
+    KEEPER_TOKEN = plugins_config["keeper"]["token"]
+except KeyError:
+    KEEPER_TOKEN = None
+
+class KeeperSecretsProvider(SecretsProvider):
+    """A secrets provider for Keeper Secrets Manager."""
+
+    slug = "keeper-secret-manager"
+    name = "Keeper Secret Manager"
+
+    class ParametersForm(BootstrapMixin, forms.Form):
+        """Parameters for Keeper Secrets Manager."""
+
+        name = forms.CharField(
+            label="Secret Name",
+            help_text="The secret's name",
+            max_length=30,
+            min_length=5,
+        )
+        uid = forms.CharField(
+            label="Secret UID",
+            help_text="The secret's uid",
+            max_length=25,
+            min_length=20,
+        )
+        token = forms.CharField(
+            label="Token",
+            widget=forms.PasswordInput,
+            help_text="The One Time Token",
+            max_length=40,
+            min_length=20,
+            initial=KEEPER_TOKEN,
+        )
+        config = forms.JSONField(
+            label="Config",
+            help_text="The JSON configuration",
```

Comment:
Any chance of giving an example of the expected structure of this field?

Reply:
https://docs.keeper.io/secrets-manager/secrets-manager/developer-sdk-library
```json
        {
            "hostname": "keepersecurity.com",
            "clientId": "ab2x3z/Acz0QFTiilm8UxIlqNLlNa25KMj=TpOqznwa4Si-h9tY7n3zvFwlXXDoVWkIs3xrMjcLGwgu3ilmq7Q==",
            "privateKey": "MLSHAgABCDEFGyqGSM49AEGCCqGSM49AwEHBG0wawIWALTARgmcnWx/DH+r7cKh4kokasdasdaDbvHmLABstNbqDwaCWhRANCAARjunta9SJdZE/LVXfVb22lpIfK4YMkJEDaFMOAyoBt0BrQ8aEhvrHN5/Z1BgZ/WpDm9dMR7E5ASIQuYUiAw0t9",
            "serverPublicKeyId": "10",
            "appKey": "RzhSIyKxbpjNu045TUrKaNREYIns+Hk9Kn8YtT+CtK0=",
            "appOwnerPublicKey": "Sq1W1OAnTwi8V/Vs/lhsin2sfSoaRfOwwDDBqoP+EO9bsBMWCzQdl9ClauDiKLXGmlmyx2xmSAdH+hlxvBRs6kU="
        }
```

How should I implement it in the code as an helper ?